pkg/api/client.go

```diff
@@ -0,0 +1,88 @@
+package api
+
+import (
+	"encoding/json"
+	"html/template"
+	"log"
+	"net/http"
+	"strconv"
+
+	"github.com/jeka2708/golang-training-enterprise/pkg/data"
+)
+
+type pageData struct {
+	Title   string
+	Clients []data.Client
+}
+
+func (a dataAPI) getAllClients(writer http.ResponseWriter, request *http.Request) {
+	clients, err := a.data.ReadAllClients()
+	pD := pageData{
+		Title:   "Список клиентов",
+		Clients: clients,
+	}
+	if err != nil {
+		_, err := writer.Write([]byte("got an error when tried to get users"))
+		if err != nil {
+			log.Println(err)
```

Comment:
You just printing an error, and not exit from function, so in step: encoding go model to json you will get an error again.
So please add writer.WriteHeader(http.StatusInternalServerError) and exit of function after printing log message.